src/plugins/ui_core_views/evaluation.ts

```diff
@@ -405,30 +956,124 @@ export class EvaluationPlugin extends UIPlugin {
     return [refFn, range, evalContext];
   }
 
+  private findCellsToCompute(mainRc: string, selfInclude: boolean = true): Iterable<string> {
+    const cellsToCompute = this.formulaDependencies.visitDeepReferences(mainRc);
+    if (!selfInclude) {
+      cellsToCompute.delete(mainRc);
+    }
+    return cellsToCompute;
+  }
+
+  private rcToCell(rc: string): Cell | undefined {
+    return this.getters.getCell(rcToCellPosition(rc));
+  }
+
+  private assertSheetHasEnoughSpaceToSpreadFormulaResult(
+    cellPosition: CellPosition,
+    matrixResult: Matrix<CellValue>
+  ) {
+    const { sheetId, col, row } = cellPosition;
+    const numberOfCols = this.getters.getNumberCols(sheetId);
+    const numberOfRows = this.getters.getNumberRows(sheetId);
+    const enoughCols = col + matrixResult.length <= numberOfCols;
+    const enoughRows = row + matrixResult[0].length <= numberOfRows;
+
+    if (enoughCols && enoughRows) {
+      return;
+    }
+
+    if (enoughCols) {
+      throw new Error(_lt("Result couldn't be automatically expanded. Please insert more rows."));
+    }
+
+    if (enoughRows) {
+      throw new Error(
+        _lt("Result couldn't be automatically expanded. Please insert more columns.")
+      );
+    }
+
+    throw new Error(
+      _lt("Result couldn't be automatically expanded. Please insert more columns and rows.")
+    );
+  }
+
   // ---------------------------------------------------------------------------
   // Export
   // ---------------------------------------------------------------------------
 
   exportForExcel(data: ExcelWorkbookData) {
     for (let sheet of data.sheets) {
-      for (const xc in sheet.cells) {
-        const position = { sheetId: sheet.id, ...toCartesian(xc) };
+      for (const rc in sheet.cells) {
```

Comment:
still a XC, not a RC

Reply:
And we're not exporting the sppreaded values of formulas that aren't exportable in excel